gcelistimg

```diff
@@ -0,0 +1,122 @@
+#
+# gceimgutils is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# gceimgutils is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with gceremoveimg. If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import logging
+import os
+import sys
+
+import gceimgutils.gceutils as utils
+import gceimgutils.gcelistimg as gcelsimg
+
+# Set up command line argument parsing
+argparse = argparse.ArgumentParser(description='List images in GCE')
+argparse.add_argument(
+    '-c', '--credentials-file',
+    dest='credentials_file_path',
+    help='Path to the service account credentials file',
+    metavar='CREDENTIALS_FILE'
+)
+argparse.add_argument(
+    '-p', '--project',
+    dest='project_name',
+    help='Project to use for image listing',
+    metavar='PROJECT_NAME',
+    required=True
+)
+# Note, one of the arguments in the group is required if --version is
+# not specified. However setting this behavior through the parser
+# also requires one argument to be specified even if --version is specified
+# This parser behavior is true even if --version and the group are part of the
+# same subgroup
+list_image_condition_group = argparse.add_mutually_exclusive_group()
+list_image_condition_group.add_argument(
+    '--image-name',
+    dest='image_name',
+    help='The image name of the image to be listed (Optional)',
+    metavar='IMAGE_NAME'
+)
+help_msg = 'An image name fragment to match the image name of the image to be '
+help_msg += 'lisetd (Optional)'
+list_image_condition_group.add_argument(
+    '--image-name-frag',
+    dest='image_name_frag',
+    help=help_msg,
+    metavar='IMAGE_NAME_FRAGMENT'
+)
+help_msg = 'A regular expression to match the image name of the image to be '
+help_msg += 'listed (Optional)'
+list_image_condition_group.add_argument(
+    '--image-name-match',
+    dest='image_name_match',
+    help=help_msg,
+    metavar='REGEX'
+)
+argparse.add_argument(
+    '--verbose',
+    action='store_true',
+    default=False,
+    dest='verbose',
+    help='Enable debug output'
+)
+argparse.add_argument(
+    '--detail',
+    action='store_true',
+    default=False,
+    dest='detail',
+    help='Enables detailed image output'
+)
+argparse.add_argument(
+    '--version',
+    action='version',
+    version=utils.get_version(),
+    help='Program version'
+)
+
+args = argparse.parse_args()
+
+logger = utils.get_logger(args.verbose)
+
+credentials_file = None
+if args.credentials_file_path:
+    credentials_file = os.path.expanduser(args.credentials_file_path)
```

Comment:
It would be helpful to move this inside of the class method. That way the logic is shared by both cli and sdk.

Reply:
@rjschwei do you want me to do this for both or leave it as is? Asking since this is mostly all of your work I copied

Reply:
For me this is a style thing. I am not nearly as picky about style as @smarlowucf is. A class appears quite heavy handed the code could be moved into utils turning 3 lines of code in every cli module into 1 line of code. If we use ec2imgutils as an example we'll save 8 lines of code.

```
credentials_file = utils.get_credentials_file(args.credentials_file_path)
```
I am fine either way.